Intermittent failing test TestInsertOrUpdateAsset in services/ticker/internal/tickerdb

The test `TestInsertOrUpdateAsset` in in the `services/ticker/internal/tickerdb` package has intermittently failed on CI. It appears to pass locally when executed a few hundred times.

Given that the failure is on time, it's possible `time.Now()` is playing a role and should be mocked within.

https://circleci.com/gh/stellar/go/7988

```
--- FAIL: TestInsertOrUpdateAsset (0.43s)
	Error Trace:	queries_asset_test.go:76
	Error:		Not equal: 
        	expected: time.Time{wall:0x4f27ac0, ext:63703668822, loc:(*time.Location)(0x12d22e0)}
        	received: time.Time{wall:0x501bd00, ext:63703668822, loc:(*time.Location)(0x12d22e0)}
        	
        	Diff:
        	--- Expected
        	+++ Actual
        	@@ -1,2 +1,2 @@
        	-(time.Time) 2019-09-09 23:33:42.083 +0000 UTC
        	+(time.Time) 2019-09-09 23:33:42.084 +0000 UTC

	Error Trace:	queries_asset_test.go:81
	Error:		Not equal: 
        	expected: time.Time{wall:0x4f27ac0, ext:63703668822, loc:(*time.Location)(0x12d22e0)}
        	received: time.Time{wall:0x501bd00, ext:63703668822, loc:(*time.Location)(0x12d22e0)}
        	
        	Diff:
        	--- Expected
        	+++ Actual
        	@@ -1,2 +1,2 @@
        	-(time.Time) 2019-09-09 23:33:42.083 +0000 UTC
        	+(time.Time) 2019-09-09 23:33:42.084 +0000 UTC

FAIL
```

--- a/services/ticker/internal/tickerdb/queries_asset_test.go
+++ b/services/ticker/internal/tickerdb/queries_asset_test.go
@@ -1,8 +1,10 @@
 package tickerdb
 
 import (
+	"reflect"
 	"testing"
 	"time"
+	"unsafe"
 
 	_ "github.com/lib/pq"
 	migrate "github.com/rubenv/sql-migrate"
@@ -50,7 +52,18 @@ func TestInsertOrUpdateAsset(t *testing.T) {
 	require.NoError(t, err)
 
 	// Creating first asset:
-	firstTime := time.Now()
+	//firstTime := time.Now()
+	firstTimeTmpl := &time.Time{}
+	{
+		firstTimeValue := reflect.ValueOf(firstTimeTmpl).Elem()
+		wallField := firstTimeValue.FieldByName("wall")
+		reflect.NewAt(wallField.Type(), unsafe.Pointer(wallField.UnsafeAddr())).Elem().SetUint(0xbf5a63f0b825c654)
+		extField := firstTimeValue.FieldByName("ext")
+		reflect.NewAt(extField.Type(), unsafe.Pointer(extField.UnsafeAddr())).Elem().SetInt(353520401)
+		locField := firstTimeValue.FieldByName("loc")
+		reflect.NewAt(locField.Type(), unsafe.Pointer(locField.UnsafeAddr())).Elem().Set(reflect.ValueOf(time.Local))
+	}
+	firstTime := *firstTimeTmpl
 	a := Asset{
 		Code:          code,
 		IssuerAccount: issuerAccount,

--- a/services/ticker/internal/tickerdb/queries_asset_test.go
+++ b/services/ticker/internal/tickerdb/queries_asset_test.go
@@ -88,13 +88,13 @@ func TestInsertOrUpdateAsset(t *testing.T) {
 	assert.Equal(t, dbIssuer.ID, dbAsset1.IssuerID)
 	assert.Equal(
 		t,
-		firstTime.Local().Truncate(time.Millisecond),
-		dbAsset1.LastValid.Local().Truncate(time.Millisecond),
+		firstTime.Local().Round(time.Millisecond),
+		dbAsset1.LastValid.Local().Round(time.Millisecond),
 	)
 	assert.Equal(
 		t,
-		firstTime.Local().Truncate(time.Millisecond),
-		dbAsset1.LastChecked.Local().Truncate(time.Millisecond),
+		firstTime.Local().Round(time.Millisecond),
+		dbAsset1.LastChecked.Local().Round(time.Millisecond),
 	)
 
 	// Creating Seconde Asset:
@@ -122,13 +122,13 @@ func TestInsertOrUpdateAsset(t *testing.T) {
 	assert.True(t, dbAsset2.LastChecked.After(firstTime))
 	assert.Equal(
 		t,
-		secondTime.Local().Truncate(time.Millisecond),
-		dbAsset2.LastValid.Local().Truncate(time.Millisecond),
+		secondTime.Local().Round(time.Millisecond),
+		dbAsset2.LastValid.Local().Round(time.Millisecond),
 	)
 	assert.Equal(
 		t,
-		secondTime.Local().Truncate(time.Millisecond),
-		dbAsset2.LastChecked.Local().Truncate(time.Millisecond),
+		secondTime.Local().Round(time.Millisecond),
+		dbAsset2.LastChecked.Local().Round(time.Millisecond),
 	)
 
 	// Creating Third Asset:
@@ -155,12 +155,12 @@ func TestInsertOrUpdateAsset(t *testing.T) {
 	assert.True(t, dbAsset3.LastChecked.Before(thirdTime))
 	assert.Equal(
 		t,
-		dbAsset2.LastValid.Local().Truncate(time.Millisecond),
-		dbAsset3.LastValid.Local().Truncate(time.Millisecond),
+		dbAsset2.LastValid.Local().Round(time.Millisecond),
+		dbAsset3.LastValid.Local().Round(time.Millisecond),
 	)
 	assert.Equal(
-		t, dbAsset2.LastValid.Local().Truncate(time.Millisecond),
-		dbAsset3.LastChecked.Local().Truncate(time.Millisecond),
+		t, dbAsset2.LastValid.Local().Round(time.Millisecond),
+		dbAsset3.LastChecked.Local().Round(time.Millisecond),
 	)
 }
 

--- a/services/ticker/internal/tickerdb/queries_asset_test.go
+++ b/services/ticker/internal/tickerdb/queries_asset_test.go
@@ -1,10 +1,8 @@
 package tickerdb
 
 import (
-	"reflect"
 	"testing"
 	"time"
-	"unsafe"
 
 	_ "github.com/lib/pq"
 	migrate "github.com/rubenv/sql-migrate"
@@ -52,18 +50,7 @@ func TestInsertOrUpdateAsset(t *testing.T) {
 	require.NoError(t, err)
 
 	// Creating first asset:
-	//firstTime := time.Now()
-	firstTimeTmpl := &time.Time{}
-	{
-		firstTimeValue := reflect.ValueOf(firstTimeTmpl).Elem()
-		wallField := firstTimeValue.FieldByName("wall")
-		reflect.NewAt(wallField.Type(), unsafe.Pointer(wallField.UnsafeAddr())).Elem().SetUint(0xbf5a63f0b825c654)
-		extField := firstTimeValue.FieldByName("ext")
-		reflect.NewAt(extField.Type(), unsafe.Pointer(extField.UnsafeAddr())).Elem().SetInt(353520401)
-		locField := firstTimeValue.FieldByName("loc")
-		reflect.NewAt(locField.Type(), unsafe.Pointer(locField.UnsafeAddr())).Elem().Set(reflect.ValueOf(time.Local))
-	}
-	firstTime := *firstTimeTmpl
+	firstTime := time.Now()
 	a := Asset{
 		Code:          code,
 		IssuerAccount: issuerAccount,

